### PR TITLE
Refs for reflection coefficients

### PR DESCRIPTION
**Pull request by steven-murray**
This PR is primarily centered around the `reflection_coefficient` module. It adds numerous bits of documentation, adding references to Raul's 2016 paper. It also adds the new `CalkitStandard` and `Calkit` classes, which are meant to capture the various parameters of a calkit. 

I'd very much appreciate a review both from @nmahesh1412 and @gitul, to make sure my understanding is correct here. You can probably mostly ignore changes in other modules -- that's just style stuff.

### CODE REVIEW

File: src/edges_cal/reflection_coefficient.py [hunk 1]
```diff
@@ -10,16 +27,18 @@ def impedance2gamma(
 ) -> float | np.ndarray:
     """Convert impedance to reflection coefficient.
 
+    See Eq. 19 of Monsale et al. 2016.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:30]: Monsale -> Monsalve

File: src/edges_cal/reflection_coefficient.py [hunk 2]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+
     Parameters
     ----------
-    gamma : float or array
+    gamma
         Reflection coefficient.
-    z0 : float or array
-        Matching impedance (?)
+    z0
+        Impedance of the match.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:60]: It should be "Reference impedance."

File: src/edges_cal/reflection_coefficient.py [hunk 3]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+
+    Obtains the instrinsic reflection coefficient from the
+    one measured at the reference plane, given a set of
+    reflection coefficients.
 
+    Parameters
+    ----------
+    s11
+        The reflection coefficient of the two-port
+        network.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:88]: change this description to:
The s11 parameter of the two-port network for the port facing the calibration plane.

(the s11 is equal to the reflection coefficient for one-port networks)

File: src/edges_cal/reflection_coefficient.py [hunk 4]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+    s12s21
+        The product of ``S12*S21`` of the two-port
+        network.
+    s22
+        The S22 of the two-port network.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:93]: change this description to:
The s22 parameter of the two-port network for the port facing the device under test (DUT).

File: src/edges_cal/reflection_coefficient.py [hunk 5]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+    gamma_ref
+        The reflection coefficient of the device
+        under test (DUT) at the reference plane.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:96]: I would say "measured at the reference plane."

File: src/edges_cal/reflection_coefficient.py [hunk 6]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+
+    Obtains the instrinsic reflection coefficient from the
+    one measured at the reference plane, given a set of
+    reflection coefficients.
+
+    Parameters
+    ----------
+    s11
+        The reflection coefficient of the two-port
+        network.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:129]: see previous comment on this.

File: src/edges_cal/reflection_coefficient.py [hunk 7]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+    s12s21
+        The product of ``S12*S21`` of the two-port
+        network.
+    s22
+        The S22 of the two-port network.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:134]: see previous comment on this.

File: src/edges_cal/reflection_coefficient.py [hunk 8]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+    gamma
+        The intrinsic reflection coefficient of the device
+        under test (DUT);.
+
+    Returns
+    -------
+    gamma_ref
+         The reflection coefficient of the DUT
+         at the reference plane.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:143]: see previous comment on this.

File: src/edges_cal/reflection_coefficient.py [hunk 9]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+def de_embed(
+    gamma_open_intr: np.ndarray,
+    gamma_short_intr: np.ndarray,
+    gamma_match_intr: np.ndarray,
+    gamma_open_meas: np.ndarray,
+    gamma_short_meas: np.ndarray,
+    gamma_match_meas: np.ndarray,
+    gamma_ref,
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    """Obtain network S-parameters from OSL standards.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:162]: ..standards as well as the intrinsic reflection coefficient of the DUT.

File: src/edges_cal/reflection_coefficient.py [hunk 10]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
-    open_C2 = 23.17e-36
-    open_C3 = -0.1597e-45
-
-    op = np.array(
-        [open_off_Zo, open_off_delay, open_off_loss, open_C0, open_C1, open_C2, open_C3]
+def input_impedance_transmission_line(
+    z0: np.ndarray, gamma: np.ndarray, length: float, z_load: np.ndarray
+) -> np.ndarray:
+    """
+    Calculate the impedance of a transmission line.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:246]: change to:
Calculates the input impedance of a terminated transmission line .

File: src/edges_cal/reflection_coefficient.py [hunk 11]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+    The standard could be open, short or load/match.
+    See the Appendix of Monsalve et al. 2016 for details.
+
+    For all parameters, 'offset' refers to the small transmission
+    line section of the standard (not an offset in the parameter).
+
+    Parameters
+    ----------
+    resistance
+        The resistance of the standard, either assumed or measured.
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:283]: change to:
The resistance of the standard termination, either ...

File: src/edges_cal/reflection_coefficient.py [hunk 12]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+        if units.get_physical_type(freq) != "frequency":
+            raise TypeError(
+                "freq must be a frequency quantity! "
+                f"Got {units.get_physical_type(freq)}"
+            )
+
+    def termination_impedance(self, freq: tp.FreqType) -> tp.OhmType:
+        """The impedance of the termination of the standard.
+
+        See Eq. 24/25 of M16 for open and short standards. The match standard
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:338]: change to:
See Eq. 22-25 of ..

File: src/edges_cal/reflection_coefficient.py [hunk 13]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+        """Reflection coefficient of the termination.
+
+        Eq. 19 of M16.
+        """
+        return impedance2gamma(self.termination_impedance(freq), 50 * units.ohm)
+
+    def lossy_characteristic_impedance(self, freq: tp.FreqType) -> tp.OhmType:
+        """Obtain the lossy characteristic impedance of the transmission line (offset).
+
+        See Eq. 20 of Monsale et al., 2016
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:361]: Monsalve

File: src/edges_cal/reflection_coefficient.py [hunk 14]
```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
...
+        gamma is the propagation constant of the transmission line (offset) and l
+        is its length. See Eq. 21 of Monsalve et al. 2016.
+        """
+        self._verify_freq(freq)
+
+        return 2 * np.pi * freq * self.offset_delay + (1 + 1j) * (
+            (self.offset_loss * self.offset_delay) / (2 * self.offset_impedance)
+        ) * np.sqrt(freq.to("GHz").value)
+
+    def gamma_offset(self, freq: tp.FreqType) -> tp.DimlessType:
```
Comment (gitul) [inline src/edges_cal/reflection_coefficient.py:380]: to be consistent with 351, shouldn't this function be named "offset_gamma" ??

**Review by gitul — COMMENTED** [review]
Very minor comments. I don't see any problem. However, I don't have experience with the way you assign units toward the end, so I trust it is correct.

File: src/edges_cal/reflection_coefficient.py [hunk 15]
```diff
@@ -31,162 +50,472 @@ def gamma2impedance(
...
+    gamma_ref
+        The reflection coefficient of the device
+        under test (DUT) measured at the reference plane.
+
+    Returns
+    -------
+    gamma
+        The intrinsic reflection coefficient of the DUT.
+
+    See Also
```
Comment (nmahesh1412) [inline src/edges_cal/reflection_coefficient.py:103]: it is gamma_embed right?

Comment (nmahesh1412) [reply src/edges_cal/reflection_coefficient.py:103]: @steven-murray

Comment (steven-murray) [reply src/edges_cal/reflection_coefficient.py:103]: Ah yes. It's fixed now, thanks!

File: src/edges_cal/reflection_coefficient.py [hunk 16]
```diff
@@ -31,162 +50,472 @@ def gamma2impedance(
...
+    Obtains the instrinsic reflection coefficient from the
+    one measured at the reference plane, given a set of
+    reflection coefficients.
 
-def gamma_shifted(s11, s12s21, s22, r):  # noqa
-    return s11 + (s12s21 * r / (1 - s22 * r))
+    Parameters
+    ----------
+    s11
+        The S11 parameter of the two-port network for the port facing the calibration
```
Comment (nmahesh1412) [inline src/edges_cal/reflection_coefficient.py:128]: @steven-murray , I am not sure if the calibration plane is the standard here or the measurement plane.

File: src/edges_cal/reflection_coefficient.py [hunk 17]
```diff
@@ -31,162 +50,472 @@ def gamma2impedance(
...
+    s22
+        The S22 of the two-port network for the port facing the device under test (DUT)
+    gamma
+        The intrinsic reflection coefficient of the device
+        under test (DUT);.
 
+    Returns
+    -------
+    gamma_ref
+         The reflection coefficient of the DUT
```
Comment (nmahesh1412) [inline src/edges_cal/reflection_coefficient.py:142]: @steven-murray here we say reference plane

File: src/edges_cal/reflection_coefficient.py [hunk 18]
```diff
@@ -31,162 +50,472 @@ def gamma2impedance(
...
+    gamma_open_meas
+        The reflection of the open standard
+        measured at port 1 as a function of frequency.
+    gamma_shrt_meas
+        The reflection of the short standard
+        measured at port 1 as a function of frequency.
+    gamma_load_meas
+        The reflection of the load standard
+        measured at port 1 as a function of frequency.
+    gamma_ref
```
Comment (nmahesh1412) [inline src/edges_cal/reflection_coefficient.py:186]: @steven-murray I thought here: gamma_xx_measured and gamma_ref are functions of frequency measured at the same point. For gamma_xx_meas we say port 1 and for gamma_ref we say reference plane

File: src/edges_cal/reflection_coefficient.py (hunk 16 above)
Comment (steven-murray) [reply src/edges_cal/reflection_coefficient.py:128]: I will use both terminologies -- @gitul suggested using "calibration"

File: src/edges_cal/reflection_coefficient.py (hunk 18 above)
Comment (steven-murray) [reply src/edges_cal/reflection_coefficient.py:186]: Hmmm, I'm not sure exactly what you mean?

**Pull request closed without merging**